src/lib/common/sensorWatchdog.cpp

```diff
@@ -1,122 +1,133 @@
-    if (_sensor_watchdog_list) {
-        sensor_watchdog_t * curr = _sensor_watchdog_list;
-        while (curr->_next) {
-            curr = curr->_next;
-        }
-        curr->_next = cfg;
-    } else {
-        _sensor_watchdog_list = cfg;
+static void _addSensorWatchdog(sensor_watchdog_t *cfg) {
+  configASSERT(cfg);
+  xSemaphoreTake(_sensor_watchdog_list_mutex, portMAX_DELAY);
+  if (_sensor_watchdog_list) {
+    sensor_watchdog_t *curr = _sensor_watchdog_list;
+    while (curr->_next) {
+      curr = curr->_next;
     }
-    xSemaphoreGive(_sensor_watchdog_list_mutex);
+    curr->_next = cfg;
+  } else {
+    _sensor_watchdog_list = cfg;
+  }
+  xSemaphoreGive(_sensor_watchdog_list_mutex);
 }
 
 static void _timerCallback(TimerHandle_t xTimer) {
-    configASSERT(xTimer);
-    configASSERT(xSemaphoreTake(_sensor_watchdog_list_mutex, 0) == pdTRUE);
-    sensor_watchdog_t * curr = _sensor_watchdog_list;
-    while(curr) {
-        if(curr->_timerHandle == xTimer){
-            configASSERT(timer_callback_handler_send_cb(_sensorWatchdogHandler, reinterpret_cast<void *>(curr), 0) == pdTRUE);
-        }
-        curr = curr->_next;
+  configASSERT(xTimer);
+  configASSERT(xSemaphoreTake(_sensor_watchdog_list_mutex, 0) == pdTRUE);
+  sensor_watchdog_t *curr = _sensor_watchdog_list;
+  while (curr) {
+    if (curr->_timerHandle == xTimer) {
+      configASSERT(timer_callback_handler_send_cb(_sensorWatchdogHandler,
+                                                  reinterpret_cast<void *>(curr), 0) == pdTRUE);
     }
-    xSemaphoreGive(_sensor_watchdog_list_mutex);
+    curr = curr->_next;
+  }
+  xSemaphoreGive(_sensor_watchdog_list_mutex);
 }
 
 static void _sensorWatchdogHandler(void *arg) {
-    configASSERT(arg);
-    sensor_watchdog_t* curr = reinterpret_cast<sensor_watchdog_t*>(arg);
-    if(curr->_logHandle) {
-        bm_fprintf(0, curr->_logHandle, "[%s] | tick: %" PRIu64 " SensorWatchdog Expired!\n", curr->_id, uptimeGetMs());
-    }
-    printf("[%s] | tick: %" PRIu64 " SensorWatchdog Expired!\n", curr->_id, uptimeGetMs());
-    configASSERT(xTimerStop(curr->_timerHandle, 0) == pdTRUE);
-    curr->_handler(arg);
-    configASSERT(xTimerReset(curr->_timerHandle, pdMS_TO_TICKS(WATCHDOG_PET_TIMEOUT_MS)) == pdTRUE);
-  if(curr->_triggerCount < curr->_max_triggers) {
-        curr->_triggerCount++;
+  configASSERT(arg);
+  sensor_watchdog_t *curr = reinterpret_cast<sensor_watchdog_t *>(arg);
+  if (curr->_logHandle) {
+    bm_fprintf(0, curr->_logHandle, "[%s] | tick: %" PRIu64 " SensorWatchdog Expired!\n",
+               curr->_id, uptimeGetMs());
+  }
+  printf("[%s] | tick: %" PRIu64 " SensorWatchdog Expired!\n", curr->_id, uptimeGetMs());
+  configASSERT(xTimerStop(curr->_timerHandle, 0) == pdTRUE);
+  curr->_handler(arg);
+  configASSERT(xTimerReset(curr->_timerHandle, pdMS_TO_TICKS(WATCHDOG_PET_TIMEOUT_MS)) ==
+               pdTRUE);
+  if (curr->_max_triggers) {
```

Comment:
This is basically the only new line - if we are set to 0, we keep resetting the timer indefinitely